renderer/components/UI/Modal.js

```diff
@@ -14,49 +16,57 @@ export const ModalOverlayStyles = () => `
   right: 0;
 `
 
-const ModalHeader = ({ hasHeader }) => (
+export const ModalHeader = styled(Panel.Header)`
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  z-index: 1;
+`
+
+const ModalLogo = ({ hasLogo }) => (
   <Flex justifyContent="space-between" px={3}>
-    {hasHeader && (
+    {hasLogo && (
       <Box color="primaryText">
         <ZapLogo height="32px" width="70px" />
       </Box>
     )}
   </Flex>
 )
 
-ModalHeader.propTypes = {
-  hasHeader: PropTypes.bool,
+ModalLogo.propTypes = {
+  hasLogo: PropTypes.bool,
 }
 
-ModalHeader.defaultProps = {
-  hasHeader: true,
+ModalLogo.defaultProps = {
+  hasLogo: true,
 }
 
 /**
- * @render react
  * @name Modal
- * @example
- * <Modal>Some content</Modal>
+ * @param {*} props Props
+ * @returns {*} Component
  */
 const Modal = props => {
-  const { children, onClose, hasClose, hasHeader, ...rest } = props
+  const { children, onClose, hasClose, hasLogo, ...rest } = props
+
   return (
     <Panel bg="primaryColor" color="primaryText">
-      <Panel.Header pt={3} px={3}>
-        {hasClose && <CloseButton onClick={onClose} />}
-        {hasHeader && <ModalHeader hasHeader={hasHeader} />}
-      </Panel.Header>
-      <Panel.Body pb={4} px={4} {...rest} css={{ 'overflow-y': 'overlay', 'overflow-x': 'hidden' }}>
-        {children}
-      </Panel.Body>
+      {(hasClose || hasLogo) && (
+        <ModalHeader pt={3} px={3}>
+          {hasClose && <CloseButton onClick={onClose} />}
```

Comment:
I wonder if `hasClose` can actually be removed at all and `onClose` could be used to determine the visibility. It's probably not worth displaying X if the handler is not set anyways. For the sake of keeping props count smaller

Reply:
I've actually done some further refactoring of this in a more recent PR so don't want to mess with this anymore here. I'll consider this change in that PR.